common/flatpak-dir.c

```diff
@@ -8639,18 +8723,21 @@ flatpak_dir_find_local_ref (FlatpakDir   *self,
                             GError      **error)
 {
   g_autofree char *local_ref = NULL;
+  g_autofree char *collection_id = NULL;
   g_autoptr(GHashTable) local_refs = NULL;
   g_autoptr(GError) my_error = NULL;
 
   if (!flatpak_dir_ensure_repo (self, NULL, error))
     return NULL;
 
-  if (!ostree_repo_list_refs (self->repo, NULL, &local_refs, cancellable, error))
+  if (!list_collection_refs_from_ostree_repo (self->repo, NULL, &local_refs,
+                                              cancellable, error))
     return NULL;
 
+  collection_id = flatpak_dir_get_remote_collection_id (self, remote);
   local_ref = find_ref_for_refs_set (local_refs, name, opt_branch,
-                                     opt_default_branch, opt_arch, kinds,
-                                     out_kind, &my_error);
+                                     opt_default_branch, opt_arch,
+                                     collection_id, kinds, out_kind, &my_error);
```

Comment:
You pass in collection_id here, but list_collection_refs_from_ostree_repo set collection_id to NULL for everything. If the remote has a collection id i don't think this will match anything.

Also, there is a pre-existing bug in this code, it passed NULL for the prefix for ostree_repo_list_refs, but it should have a prefix like `$remote:` so that only apps from the matched remote are listed, no?

Reply:
I think you're right. Pushed a fixup